nuxeo-wopi/src/main/java/org/nuxeo/wopi/WOPIServiceImpl.java

```diff
@@ -91,6 +122,22 @@ public void start(ComponentContext context) {
         log.debug("Registered old proof key: {}", oldProofKey);
     }
 
+    protected byte[] fetchDiscovery() {
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        try (CloseableHttpClient httpClient = httpClientBuilder.build()) {
+            HttpGet request = new HttpGet(discoveryURL);
+            try (CloseableHttpResponse response = httpClient.execute(request)) {
+                try (InputStream is = response.getEntity().getContent()) {
+                    return IOUtils.toByteArray(is);
+                }
+            }
+        } catch (IOException e) {
+            log.error("Error while fetching WOPI discovery", e);
+            log.debug(e, e);
```

Comment:
Did you mean to log the stack trace in both error and debug?

Reply:
Nop, only the exception message in `error`.
So something like `log.error("Error while fetching WOPI discovery: {}", e)` will probably be better.

Reply:
Check that it works then because I thought a `Throwable` in last position was used for stack trace.

Reply:
Let's be safe with `log.error("Error while fetching WOPI discovery: {}", e.getMessage())` :)

Reply:
Needs to be checked indeed, afaik `Throwable` stacktrace is printed if there's no parameter (`{}`) to handle the parameter.